plugins/inputs/beat/beat.go

```diff
@@ -0,0 +1,278 @@
+  url = "http://127.0.0.1:5066"
+
+  ## Enable collection of the Beat stats
+  collect_beat_stats = true
+
+  ## Enable the collection if Libbeat stats
+  collect_libbeat_stats = true
+
+  ## Enable the collection of OS level stats
+  collect_system_stats = false
+
+  ## Enable the collection of Filebeat stats
+  collect_filebeat_stats = true
+
+  ## HTTP method
+  # method = "GET"
+
+  ## Optional HTTP headers
+  # headers = {"X-Special-Header" = "Special-Value"}
+
+  ## Override HTTP "Host" header
+  # host_header = "logstash.example.com"
+
+  ## Timeout for HTTP requests
+  timeout = "5s"
+
+  ## Optional HTTP Basic Auth credentials
+  # username = "username"
+  # password = "pa$$word"
+
+  ## Optional TLS Config
+  # tls_ca = "/etc/telegraf/ca.pem"
+  # tls_cert = "/etc/telegraf/cert.pem"
+  # tls_key = "/etc/telegraf/key.pem"
+  ## Use TLS but skip chain & host verification
+  # insecure_skip_verify = false
+`
+
+const description = "Read metrics exposed by Beat"
+
+const suffixInfo = "/"
+const suffixStats = "/stats"
+
+type BeatInfo struct {
+	Beat     string `json:"beat"`
+	Hostname string `json:"hostname"`
+	Name     string `json:"name"`
+	UUID     string `json:"uuid"`
+	Version  string `json:"version"`
+}
+
+type BeatStats struct {
+	Beat     map[string]interface{} `json:"beat"`
+	FileBeat interface{}            `json:"filebeat"`
+	Libbeat  interface{}            `json:"libbeat"`
+	System   interface{}            `json:"system"`
+}
+
+type Beat struct {
+	URL string `toml:"url"`
+
+	CollectBeatStats     bool `toml:"collect_beat_stats"`
+	CollectLibbeatStats  bool `toml:"collect_libbeat_stats"`
+	CollectSystemStats   bool `toml:"collect_system_stats"`
+	CollectFilebeatStats bool `toml:"collect_filebeat_stats"`
+
+	Username   string            `toml:"username"`
+	Password   string            `toml:"password"`
+	Method     string            `toml:"method"`
+	Headers    map[string]string `toml:"headers"`
+	HostHeader string            `toml:"host_header"`
+	Timeout    internal.Duration `toml:"timeout"`
+
+	tls.ClientConfig
+	client *http.Client
+}
+
+func NewBeat() *Beat {
+	return &Beat{
+		URL:                  "http://127.0.0.1:5066",
+		CollectBeatStats:     true,
+		CollectLibbeatStats:  true,
+		CollectSystemStats:   true,
+		CollectFilebeatStats: true,
+		Method:               "GET",
+		Headers:              make(map[string]string),
+		HostHeader:           "",
+		Timeout:              internal.Duration{Duration: time.Second * 5},
+	}
+}
+
+func (beat *Beat) Description() string {
+	return description
+}
+
+func (beat *Beat) SampleConfig() string {
+	return sampleConfig
+}
+
+// createHttpClient create a clients to access API
+func (beat *Beat) createHttpClient() (*http.Client, error) {
+	tlsConfig, err := beat.ClientConfig.TLSConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: tlsConfig,
+		},
+		Timeout: beat.Timeout.Duration,
+	}
```

Comment:
How about e.g. proxy settings?

Reply:
Couldn't find many other plugins that support proxies, was there something about this one in particular that you thought necessitated it?

Reply:
The http plugin supports proxies for examples. The reason is that I had quite some software _not_ accepting proxies in the past and that might be a nightmare in enterprise environments. I don't think adding those is super hard either.

Reply:
From what I can tell, the http plugin is the only other plugin that supports proxies. I could see adding it if there's enough prevailing need or consistency with the other plugins that support it. Seems like something that could be easily added if necessary.

Reply:
I take your word on this! ;-) If you really don't want to do it now, mark this as resolved...